core/example/benchmark/Cabana_CommPerformance.cpp

```diff
@@ -0,0 +1,573 @@
+            for ( int i = -1; i < 2; ++i )
+                if ( !(i==0 && j==0 && k==0) )
+                {
+                    std::vector<int> ncr = { cart_rank[0] + i,
+                                             cart_rank[1] + j,
+                                             cart_rank[2] + k };
+                    int nr;
+                    MPI_Cart_rank( cart_comm, ncr.data(), &nr );
+                    neighbor_ranks.push_back( nr );
+                }
+    std::sort( neighbor_ranks.begin(), neighbor_ranks.end() );
+    auto unique_end =
+        std::unique( neighbor_ranks.begin(), neighbor_ranks.end() );
+    neighbor_ranks.resize( std::distance(neighbor_ranks.begin(),unique_end) );
+
+    // Define the aosoa.
+    using comm_memory_space = typename CommDevice::memory_space;
+    using data_memory_space = typename DataDevice::memory_space;
+    using member_types = Cabana::MemberTypes<double[3],
+                                             double[3],
+                                             double,
+                                             int>;
+    using aosoa_type = Cabana::AoSoA<member_types,data_memory_space>;
+
+    // Get the byte size of a particle
+    using tuple_type = Cabana::Tuple<member_types>;
+    int bytes_per_particle = sizeof( tuple_type );
+
+    // Number of runs in the test loops.
+    int num_run = 10;
+
+    // Fraction of particles on each rank that will be communicated to the
+    // neighbors. We will sweep through these fractions to get an indicator of
+    // performance as a function of message size.
+    std::vector<double> comm_fraction = { 0.0001,
+                                          0.001,
+                                          0.005,
+                                          0.01,
+                                          0.05,
+                                          0.10,
+                                          0.5 };
+    int num_fraction = comm_fraction.size();
+
+    // Number of bytes we will send to each neighbor.
+    std::vector<std::size_t> comm_bytes( num_fraction );
+
+    // DISTRIBUTOR
+    // -----------
+
+    // Create distributor timers.
+    ParallelTimer distributor_fast_create(
+        comm, test_prefix+"distributor_fast_create", num_fraction );
+    ParallelTimer distributor_slow_create(
+        comm, test_prefix+"distributor_slow_create", num_fraction );
+    ParallelTimer distributor_aosoa_migrate(
+        comm, test_prefix+"distributor_aosoa_migrate", num_fraction );
+    ParallelTimer distributor_slice_migrate(
+        comm, test_prefix+"distributor_slice_migrate", num_fraction );
+
+    // Loop over comm fractions.
+    for ( int fraction = 0; fraction < num_fraction; ++fraction )
+    {
+        // Create the migrate distribution in the data memory space. This is
+        // where it would likely be created.
+        int num_send = num_particle * comm_fraction[fraction];
+        int send_per_neighbor = num_send / 26;
```

Comment:
This `/26` then `*26` strikes me as a little odd ? :p

Reply:
That was a way to cheat and get a number that was evenly divisible by 26

Reply:
Add a comment to capture the intent

Reply:
I have added a comment discussing intent